google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java

```diff
@@ -504,12 +562,33 @@ public BlobWriteChannel writer(BlobInfo blobInfo, BlobWriteOption... options) {
 
   @Override
   public BlobWriteChannel writer(URL signedURL) {
-    return new BlobWriteChannel(getOptions(), signedURL);
+    ExceptionHandler forResumableUploadSessionCreate =
+        retryAlgorithmManager.getForResumableUploadSessionCreate(
+            Collections
+                .emptyMap()); // TODO: is it possible to know if a signed url is configured to have
+    // a constraint which makes it idempotent?
```

Comment:
o, this is a really good call out. You can in theory generate many sessions with one signed URL IIRC. As far as making it idempotent, yes, you can use precondition headers for XML API: https://cloud.google.com/storage/docs/xml-api/put-object#request-headers